### PR TITLE
Omit erroneous overview elements

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

This filters the return array of the Safe overview endpoint (`/v1/safes`), removing any elements that threw. This prevents one thrown retrieval of a `SafeOverview` from blocking others.

## Changes

- Change request from `Promise.all` to `Promise.allSettled`
- Filter out `'rejected'` elements, logging if existent

### CODE REVIEW

File: src/routes/safes/safes.controller.overview.spec.ts [hunk 1]
```diff
@@ -1036,22 +1036,181 @@ describe('Safes Controller Overview (Unit)', () => {
   });
 
   describe('Validation', () => {
-    it('forwards error responses from the Config Service', async () => {
-      const chain = chainBuilder().with('chainId', '10').build();
-      const safeInfo = safeBuilder().build();
+    it('omits erroroneous Safes on chains returning an error from the Config Service', async () => {
```
Comment (hectorgomezv) [inline src/routes/safes/safes.controller.overview.spec.ts:1039]: Nit: `erroneous`

File: src/routes/safes/safes.controller.overview.spec.ts [hunk 2]
```diff
@@ -1060,31 +1219,196 @@ describe('Safes Controller Overview (Unit)', () => {
...
+              awaitingConfirmation: expect.any(Number),
+            },
+          ]),
+        );
     });
 
-    it('forwards error responses from the Transaction Service', async () => {
-      const chain = chainBuilder().with('chainId', '10').build();
-      const safeInfo = safeBuilder().build();
+    it('omits erroroneous Safes from the Transaction Service', async () => {
```
Comment (hectorgomezv) [inline src/routes/safes/safes.controller.overview.spec.ts:1266]: Nit: `erroneous`

File: src/routes/safes/safes.controller.overview.spec.ts [hunk 3]
```diff
@@ -1094,46 +1418,216 @@ describe('Safes Controller Overview (Unit)', () => {
...
+        );
     });
 
-    it('throw a 500 if validation of the Transaction Service return Promise.rejects', async () => {
-      const chain = chainBuilder().with('chainId', '10').build();
-      const safeInfo = safeBuilder().build();
-      const tokenAddress = faker.finance.ethereumAddress();
-      const secondTokenAddress = faker.finance.ethereumAddress();
-      const transactionApiBalancesResponse = [
+    it('omits erroroneous Safes validation of the Transaction Service return Promise.rejects', async () => {
```
Comment (hectorgomezv) [inline src/routes/safes/safes.controller.overview.spec.ts:1465]: Nit: `erroneous`

**Comment by hectorgomezv** [comment]
Superb job! I've left a tiny nit thing only.

File: src/routes/safes/safes.controller.overview.spec.ts (hunk 1 above)
Comment (iamacook) [reply src/routes/safes/safes.controller.overview.spec.ts]: Fixed in 56fee09f357b6eab83f577eebe3fd43ed21db7fb.

File: src/routes/safes/safes.controller.overview.spec.ts (hunk 2 above)
Comment (iamacook) [reply src/routes/safes/safes.controller.overview.spec.ts]: Fixed in 56fee09f357b6eab83f577eebe3fd43ed21db7fb.

File: src/routes/safes/safes.controller.overview.spec.ts (hunk 3 above)
Comment (iamacook) [reply src/routes/safes/safes.controller.overview.spec.ts]: Fixed in 56fee09f357b6eab83f577eebe3fd43ed21db7fb.

**Review by hectorgomezv — APPROVED** [review]

File: src/routes/safes/safes.service.ts [hunk 4]
```diff
@@ -176,6 +179,20 @@ export class SafesService {
         );
       }),
     );
+
+    const safeOverviews: Array<SafeOverview> = [];
+
+    for (const safeOverview of settledOverviews) {
+      if (safeOverview.status === 'rejected') {
+        this.loggingService.error(
```
Comment (fmrsabino) [inline src/routes/safes/safes.service.ts:187]: I would change this to `warn` level as I see this message as something that we need to be aware of but doesn't stop the endpoint from working.

**Review by fmrsabino — APPROVED** [review]

File: src/routes/safes/safes.service.ts (hunk 4 above)
Comment (iamacook) [reply src/routes/safes/safes.service.ts]: Changed in 15d1d5fd456a45d446a767856d230dd477705079.

**Review by fmrsabino — APPROVED** [review]

**Pull request merged**